src/aiida/tools/dumping/utils.py

```diff
@@ -0,0 +1,72 @@
+###########################################################################
+# Copyright (c), The AiiDA team. All rights reserved.                     #
+# This file is part of the AiiDA code.                                    #
+#                                                                         #
+# The code is hosted on GitHub at https://github.com/aiidateam/aiida-core #
+# For further information on the license, see the LICENSE.txt file        #
+# For further information please visit http://www.aiida.net               #
+###########################################################################
+"""Utility functions for dumping features."""
+
+from __future__ import annotations
+
+import logging
+import shutil
+from pathlib import Path
+
+__all__ = ['validate_make_dump_path']
+
+logger = logging.getLogger(__name__)
+
+
+def validate_make_dump_path(
+    path_to_validate: Path,
+    overwrite: bool = False,
+    incremental: bool = True,
+    safeguard_file: str = '.aiida_node_metadata.yaml',
+    enforce_safeguard: bool = True,
+) -> Path:
+    """Create default dumping directory for a given process node and return it as absolute path.
+
+    :param validate_path: Path to validate for dumping.
+    :param safeguard_file: Dumping-specific file to avoid deleting wrong directory.
+        Default: `.aiida_node_metadata.yaml`
+    :return: The absolute created dump path.
+    """
+
+    if overwrite and incremental:
+        raise ValueError('Both overwrite and incremental set to True. Only specify one.')
+
+    # Handle existing directory
+    if path_to_validate.is_dir():
```

Comment:
Don't we need to also handle the case that it is file? In this case it exists, so mkdir later on should fail, but this if case is never entered.

Reply:
Yes, good catch!